packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -259,6 +262,10 @@ class CurlyComponentManager {
   }
 
   didCreateElement({ component, classRef, environment }, element, operations) {
+    if (environment.isInteractive) {
+      component.trigger('willRender');
+    }
```

Comment:
Actually did some more investigation. Moved this and updated the tests accordingly.